core/src/rust/filodb_core/src/profile.rs

```diff
@@ -0,0 +1,112 @@
+//! Helpers for profiling / testing
+
+#[cfg(feature = "dhat-heap")]
+use std::sync::Mutex;
+
+use jni::{
+    objects::JClass,
+    sys::{jlong, jstring},
+    JNIEnv,
+};
+
+use crate::{exec::jni_exec, state::IndexHandle};
+
+/// Get cache stats info
+#[no_mangle]
+pub extern "system" fn Java_filodb_core_memstore_TantivyNativeMethods_00024_dumpCacheStats(
+    mut env: JNIEnv,
+    _class: JClass,
+    handle: jlong,
+) -> jstring {
+    jni_exec(&mut env, |env| {
+        let index = IndexHandle::get_ref_from_handle(handle);
+
+        let (column_hits, column_misses) = index.column_cache.stats();
+        let (query_hits, query_misses) = index.query_cache_stats();
+
+        let output = format!(
```

Comment:
While it's ok to log, the value will determine the changes we will make to ``column-cache-count``. Does it make sense to have a gauge for this as a metric (need not be part of this PR)?

Reply:
Let me think about this for a future PR.  In practice with this cache I'm not sure we'll ever tune it - 1000 entries is not much RAM and we don't have that many actual columns to cache / segments even as we grow.

Reply:
I thought about this some more - what I'd like to do is a follow up PR that adds the following metrics:

* Cache hit rates for both caches
* Ingestion commit duration

I'll make sure these are in before we merge to develop.